libtiledbsoma/src/soma/managed_query.cc

```diff
@@ -1239,30 +1239,23 @@ ManagedQuery::_extend_and_evolve_schema_with_details(
     // Find any new enumeration values
     std::vector<ValueType> enum_values_to_add;
     for (auto enum_val : enum_values_in_write) {
-        bool is_new_value = false;
-
-        if (std::isnan(enum_val)) {
-            // Special case: check if the value is NaN. We cannot use std::find
-            // below because NaN != NaN
-            for (auto existing_val : enum_values_existing) {
-                if (std::isnan(existing_val)) {
-                    is_new_value = true;
-                    break;
-                }
-            }
-
-        } else {
-            // Typical case: use std::find to check if the value already exists
-            if (std::find(
-                    enum_values_existing.begin(),
-                    enum_values_existing.end(),
-                    enum_val) != enum_values_existing.end()) {
-                is_new_value = true;
-            }
-        }
+        // Check if the value already exists using bitwise comparison.
+        // - We cannot use std::find because NaN != NaN
+        // - We cannot use std::isnan because multiple bit representations
+        //   exist for NaN. For example, in float32 (little-endian), both
+        //   0x7fc00000 and 0x7fc00100 represent NaN
```

Comment:
Actually we _totally could_ use `std::isnan`. That's not a problem. The problem is that core uses bit-equality to check against a particular NaN value, while `std::isnan` would check against all possible NaN values. In an ideal world, the latter would (IMO) be better. But as @bkmartinjr pointed out, let's stick with core's bit-equality checking, solely for consistency with core.